### PR TITLE
tdd

### PR DESCRIPTION
**Pull request by Kgomotso196**
# PROJECT TO DOs
### IN THIS PROJECT WE:
1. Wrote unit tests using Rspec.
2. Understood an existing codebase in a way that allows for adding tests.
3. Wrote Rspec unit tests with mocks.
4. Understood the TDD concept.
5. Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
6. Made sure that we used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
7. Made sure that we documented our work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
8. Followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
9. Created a class called `Solver`.
10. Created a method called `factorial` that takes one argument, an integer `N`, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number.
11. Created a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if word is "hello" it returns "olleh").
12. Created a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      it 'returns Fizzbuzz when given 15' do
+        expect(Solver.new.fizzbuzz(15)).to eq('fizzbuzz')
+      end
+      it 'returns Fizz when given 3' do
+        expect(Solver.new.fizzbuzz(3)).to eq('fizz')
+      end
+      it 'returns Buzz when given 5' do
+        expect(Solver.new.fizzbuzz(5)).to eq('buzz')
+      end
+    end
```
Comment (omar-muhamad) [inline spec/solver_spec.rb]: - [ ] In TDD tests are supposed to cover all edge cases, As per this [requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=Any%20other%20case%2C%20return%20N%20as%20a%20string%20(e.g.%20say%20N%20is%207%20then%20return%20%227%22)), Please add a test in case of a regular number like 7 `not divisible by 3 or 5` the method suppose to return it as it is.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Kgomotso196 and @kimitaWanjohi,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by omar-muhamad** [comment]
Hi @Kgomotso196 and @kimitaWanjohi,
I required the changes above but I wrongly approved the project from My reviewer dashboard so You will see it on your dashboard as approved.
Please don't take any action until the Code Reviewers team fixes that issue.
and don't worry it will be fixed soon.
Have a great day.

File: spec/solver_spec.rb (hunk 1 above)
Comment (Kgomotso196) [reply spec/solver_spec.rb]: @Omar-Muhamad  Thanks for your kind feedback. Working on it.

**Review by sumairq — APPROVED** [review]
### Hi Team , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good job implementing changes
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
